### PR TITLE
Remove usage of the deprecated NonRoot feature gate

### PR DESCRIPTION
**Pull request by orelmisan**
**What this PR does / why we need it**:
The `NonRoot` feature gate was deprecated in PR #8563.

The presence or absence of this feature gate does not change KubeVirt's behavior.

The `NonRoot` feature gate is not present in the CI environment since kubevirt/project-infra#2418:
- Remove the NonRoot feature e2e test.
- Remove two code paths that were dependent on the presence of this feature gate in the storage e2e tests.
- Remove a leftover code path in `manifest-templator`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
~~This PR depends on #10608, please skip the first commit.~~

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
Remove leftover NonRoot feature gate
```

### CODE REVIEW

**Review by fossedihelm — COMMENTED** [review]
Thanks @orelmisan! Nice to see this go forward!
I think we can also drop `NonRootExperimental` FG. 
We probably should have done this long ago.

File: pkg/virt-config/feature-gates.go
Comment (fossedihelm) [reply pkg/virt-config/feature-gates.go:48]: My bad! I just noticed https://github.com/kubevirt/kubevirt/pull/10582. Sorry for the noise

File: tests/storage/storage.go
Comment (orelmisan) [reply tests/storage/storage.go:734]: I raised both storage test changes in the SIG-storage meeting today.
This code was not essential for the test to pass because of changes in KubeVirt/CDI.
@mhenriks do you want me to add additional details to the commit messages?

File: pkg/virt-config/feature-gates.go
Comment (EdDev) [inline pkg/virt-config/feature-gates.go:90]: Some time ago a policy to deprecate a FG was created [1], which required old FGs to be left around, warning users that it is no longer in use.

Therefore, this is not exactly "dead code".
To remove this, we need to explain how it fits the policy or to update the policy.

Generally speaking, I am in favor to remove it, making sure the policy has such a path (e.g. after 2 versions, remove it completely but make sure it does not break old configurations).

[1] https://github.com/kubevirt/kubevirt/pull/8047

**Review by EdDev — COMMENTED** [review]
The last commit may be problematic, see inline comment.

As FG config have user-facing API, it will be good to emphasize that such changes will not break upgrades (e.g. if a cluster with this FG configured is upgrade, it will be silently ignored and not blow up). I think at the moment it is intentionally treated and a deprecation message generated.

File: tests/storage/storage.go
Comment (akalenyu) [reply tests/storage/storage.go:734]: @xpivarc Can't remember why we needed this. Doesn't virt-handler do this for us

Comment (xpivarc) [reply tests/storage/storage.go:734]: I believe this was just an initial limitation. See https://github.com/kubevirt/kubevirt/pull/6041 where this was added and https://github.com/kubevirt/kubevirt/pull/6240 where it should have been dropped.

**Comment by xpivarc** [comment]
/approve
Please close the conversation with Ed before merging, otherwise, this looks good.

File: pkg/virt-config/feature-gates.go
Comment (orelmisan) [reply pkg/virt-config/feature-gates.go]: Removed the templator leftover.

File: pkg/virt-config/feature-gates.go
Comment (orelmisan) [reply pkg/virt-config/feature-gates.go]: I dropped the commit which removed the `NonRoot` feature gate.

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/0987f00cf2f258673acedb9103ba28e991439d9f..3d7f0880dbf342ec301ccaa82bcf65622b0d9edb): Rebase
Removed the NonRoot feature e2e test
Removed the manifest templator leftover
Dropped the removal of the NonRoot feature gate

**Comment by orelmisan** [comment]
@EdDev @fossedihelm could you please have another look?

**Comment by mhenriks** [comment]
changes to storage tests look good

**Comment by fossedihelm** [comment]
@orelmisan Thanks!
/lgtm
/hold
@EdDev feel free to unhold if everything is ok for you!

**Review by EdDev — APPROVED** [review]
Thanks!

/unhold

**Pull request merged**
